ui/pages/permissions-connect/connect-page/connect-page.tsx

```diff
@@ -116,11 +120,20 @@ export const ConnectPage: React.FC<ConnectPageProps> = ({
     request.permissions,
   );
 
+  const requestedCaip25CaveatValueWithExistingPermissions =
+    request.existingCaveat
+      ? mergeCaip25CaveatValues(
```

Comment:
we might be able to use the chain-agnostic-permission helper here. maybe `mergeInternalScopes`?